### PR TITLE
doc: add explanation about extended recursive behavior

### PR DESCRIPTION
**Pull request by kenhys**

### CODE REVIEW

File: doc/source/reference/commands/io_flush.rst [hunk 1]
```diff
@@ -279,6 +279,81 @@ isn't specified:
...
+   * - ``flush[(anonymous:column:var_size)]``
+     - The internal object metadata (builtin types, token filter and so on) are flushed.
+
+       This is variable sized column which stores so called ``spec`` internally.
+
+   * - ``flush[(anonymous:table:hash_key)]``
+
+     - The internal configuration objects (which is set by ``config_set``) are flushed.
+
+   * - ``flush[(anonymous:column:var_size)]``
```
Comment (komainu8) [inline doc/source/reference/commands/io_flush.rst]: This log message is same https://github.com/groonga/groonga/pull/938/files#diff-acf8eb35060c746abd34f25d4ee37cc5R339.
How do we distinguish these log message? 
Can we separate by output order?

File: doc/source/reference/commands/io_flush.rst [hunk 2]
```diff
@@ -279,6 +279,81 @@ isn't specified:
...
+   * - Log
+     - Description
+
+   * - ``flush[(anonymous:table:dat_key)]``
+     - The internal object names in DB are flushed. If ``GRN_DB_KEY=pat`` is set, ``TABLE_PAT_KEY`` is used instead.
+
+   * - ``flush[(anonymous:column:var_size)]``
+     - The internal object metadata (builtin types, token filter and so on) are flushed.
+
+       This is variable sized column which stores so called ``spec`` internally.
```
Comment (komainu8) [inline doc/source/reference/commands/io_flush.rst]: What is `spec`?
Is `spec` the internal object metadata?
If so, how about explaining as below?

```
This is a variable sized column which stores the above internal object metadata.
```

File: doc/source/reference/commands/io_flush.rst [hunk 3]
```diff
@@ -279,6 +279,81 @@ isn't specified:
...
+* A table of related index column (There is source column in target TABLE_NAME)
+
+It is useful not to forget flushing related objects.
+
+For example, ``--recursive yes`` is specified for ``TABLE_NAME``, this
+option executes equivalent to the following commands internally.
+
+- Flush table and its columns::
+
+    io_flush --target_name TABLE_NAME
```
Comment (komainu8) [inline doc/source/reference/commands/io_flush.rst]: a default value of `recursive ` is `yes`.
Therefore, I think if we execute this command, flush not only target object and child objects but also related objects.

File: doc/source/reference/commands/io_flush.rst (hunk 1 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: I've fixed to add explanation explicitly. (which one is explained)

File: doc/source/reference/commands/io_flush.rst (hunk 2 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: Thanks, It should be as suggested.

File: doc/source/reference/commands/io_flush.rst (hunk 3 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: I've missed it, so fixed it.

**Comment by kenhys** [comment]
@komainu8  I've fixed based on your feedback. Could you check it again, please?

File: doc/source/reference/commands/io_flush.rst [hunk 4]
```diff
@@ -279,6 +279,81 @@ isn't specified:
...
+* A table of related index column (There is source column in target TABLE_NAME)
+
+It is useful not to forget flushing related objects.
+
+For example, ``--recursive yes`` (default) is specified for ``TABLE_NAME``, this
+option executes equivalent to the following commands internally.
+
+- Flush table and its columns::
+
+    io_flush --target_name TABLE_NAME --recursive no
```
Comment (komainu8) [inline doc/source/reference/commands/io_flush.rst]: I think if we execute this command, we can not but flush a table.
If we flush the table and its columns, we also execute a below command.

```
io_flush --target_name TABLE_NAME.COLUMN_NAME1 --recursive no
io_flush --target_name TABLE_NAME.COLUMN_NAME2 --recursive no
io_flush --target_name TABLE_NAME.COLUMN_NAME3 --recursive no
.
.
.
```

Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: Thanks!
I've fixed it.

**Comment by kenhys** [comment]
@komainu8 I've rewrite a document about `recursive=dependent` again.

**Comment by kenhys** [comment]
@kou could you review this PR, please?

File: doc/source/reference/commands/io_flush.rst [hunk 5]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+
+     - Use ``--recursive=dependent`` to flush target table and its columns,
+       referenced tables and tables of corresponding index columns and corresponding index
+       columns at once::
+
+         io_flush --target_name TABLE_NAME --recursive=dependent
+
+       Then flush database::
+
+         io_flush --recursive no
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: Is this needed?

File: doc/source/reference/commands/io_flush.rst [hunk 6]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+   * - :doc:`table_create`
+     - Target table and database.
+     - Table::
+
+         io_flush --target_name TABLE_NAME
+
+       Database::
+
+         io_flush --recursive no
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: Is this needed?

File: doc/source/reference/commands/io_flush.rst [hunk 7]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+         io_flush --target_name TABLE_NAME --recursive=dependent
+
+       Then flush database::
+
+         io_flush --recursive no
+   * - :doc:`table_create`
+     - Target table and database.
+     - Table::
+
+         io_flush --target_name TABLE_NAME
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: Can we use `--recursive dependent`?

File: doc/source/reference/commands/io_flush.rst [hunk 8]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+     - Target table and its columns.
+
+       If there are one or more reference columns in these columns,
+       referenced tables are also flush targets.
+
+       If there are one or more indexed columns in these columns,
+       tables of corresponding index columns and corresponding index
+       columns are also flush targets.
+
+     - Use ``--recursive=dependent`` to flush target table and
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: `=` is invalid.

File: doc/source/reference/commands/io_flush.rst [hunk 9]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+       If there are one or more reference columns in these columns,
+       referenced tables are also flush targets.
+
+       If there are one or more indexed columns in these columns,
+       tables of corresponding index columns and corresponding index
+       columns are also flush targets.
+
+       Database is also flush target.
+
+     - Use ``--recursive=dependent`` to flush target table and its columns,
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: `=` is invalid.

File: doc/source/reference/commands/io_flush.rst [hunk 10]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+         io_flush --recursive no
+   * - :doc:`column_create`
+     - Target column and database.
+     - Table::
+
+         io_flush --target_name TABLE_NAME.COLUMN_NAME
+
+       Database::
+
+         io_flush --recursive no
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: Is this needed?

File: doc/source/reference/commands/io_flush.rst [hunk 11]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+     - Database::
+
+         io_flush --recursive no
+   * - :doc:`plugin_register` and :doc:`plugin_unregister`
+     - Database.
+     - Database::
+
+         io_flush --recursive no
+
+If Groonga is 9.0.1 or earlier ( ``recursive=dependent`` is not available ), flush
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: `=` is invalid.

File: doc/source/reference/commands/io_flush.rst [hunk 12]
```diff
@@ -279,6 +372,81 @@ isn't specified:
 .. include:: ../../example/reference/commands/io_flush/recursive_default.log
 .. io_flush
 
+Since 9.0.2, ``recursive=dependent`` is added to flush not only target object
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: `=` is invalid.

File: doc/source/reference/commands/io_flush.rst [hunk 13]
```diff
@@ -279,6 +372,81 @@ isn't specified:
...
+and child objects, but also related objects. The related objects are:
+
+* A referenced table
+* A related index column (There is source column in target TABLE_NAME)
+* A table of related index column (There is source column in target TABLE_NAME)
+
+It is useful not to forget flushing related objects.
+
+For example, ``--recursive=dependent`` is specified for ``TABLE_NAME``, this
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: `=` is invalid.

File: doc/source/reference/commands/io_flush.rst [hunk 14]
```diff
@@ -279,6 +372,81 @@ isn't specified:
...
+
+- Flush table and its columns::
+
+    io_flush --target_name TABLE_NAME --recursive yes
+
+- Flush a referenced table::
+
+    io_flush --target_name REFERENCED_TABLE_NAME --recursive no
+
+- Flush a related index column (There is source column in TABLE_NAME)::
```
Comment (kou) [inline doc/source/reference/commands/io_flush.rst]: Should we use ` ``TABLE_NAME`` ` here?

File: doc/source/reference/commands/io_flush.rst (hunk 5 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: @kou
Maybe I'm understand wrongly, I thought that `io_flush` with `--target_name` needs flush database separately.
I'll fix it.

Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: Fixe

File: doc/source/reference/commands/io_flush.rst (hunk 6 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: It should use `--recursive dependent` and remove it.

File: doc/source/reference/commands/io_flush.rst [hunk 15]
```diff
@@ -87,7 +92,93 @@ You can flush all changes in memory to disk with no arguments:
...
+       tables of corresponding index columns and corresponding index
+       columns are also flush targets.
+
+       Database is also flush target.
+
+     - Use ``--recursive=dependent`` to flush target table and its columns,
+       referenced tables and tables of corresponding index columns and corresponding index
+       columns at once::
+
+         io_flush --target_name TABLE_NAME --recursive=dependent
```
[the remark this answers is not in the document]
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: fixed

File: doc/source/reference/commands/io_flush.rst (hunk 14 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: fixed

File: doc/source/reference/commands/io_flush.rst (hunk 10 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: needless if --recursive dependent is used.

Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: I'll fix

File: doc/source/reference/commands/io_flush.rst (hunk 7 above)
Comment (kenhys) [reply doc/source/reference/commands/io_flush.rst]: I've got it.

**Pull request merged**